lldb/source/Plugins/SymbolLocator/Debuginfod/SymbolLocatorDebuginfod.cpp

```diff
@@ -54,6 +55,34 @@ class PluginProperties : public Properties {
     return urls;
   }
 
+  llvm::Expected<llvm::StringRef> GetCachePath() {
```

Comment:
Is it always safe to return a StringRef here? It looks fine at first sight, but this code also doesn't look particularly hot, so playing it safe and returning a `std::string` avoids having to reason about the string's lifetime.

Reply:
String storage lives 40 lines lower (line 97) because, yeah, this bit me. The Debuginfod library takes a StringRef, and I found this was the most consistent way to deal with it. If you'd prefer me to switch this and cons up a StringRef before the API, I'm okay with that (just let me know in here)

Reply:
Actually, I do think it's wrong after all. On line 69 you return a `StringRef` to a temporary on the stack (created on line 66). This must be a std::string.

Reply:
D'oh. You're right. I'll hoist that StringRef up and switch the function to return a string. Good catch!